script/draft-release/version.ts

```diff
@@ -0,0 +1,57 @@
+import { inc, parse } from 'semver'
+
+import { Channel } from './channel'
+
+export function getNextVersionNumber(
+  version: string,
+  channel: Channel
+): string {
+  const parsed = parse(version)
+
+  if (parsed == null) {
+    throw new Error(`Unable to parse input '${version}' into version`)
+  }
+
+  if (channel === 'production') {
+    if (parsed.prerelease.some(p => p.startsWith('beta'))) {
```

Comment:
Could we extract the logic from this condition body into a function, `tryIncremenetProductionVersion: (version: SemVer) => string` (or similar name)?